src/qp_klp/Protocol.py

```diff
@@ -22,6 +25,48 @@ class Protocol():
      initialization.
     """
     protocol_type = PROTOCOL_NAME_NONE
+    # this value was selected by looking at all the successful NuQC/SPP jobs,
+    # the max sequeces were: 712,497,596
+    MAX_READS = 720000000
+
+    def subsample_reads(self):
+        if self.assay_type == 'Amplicon':
+            return
+
+        df = pd.read_csv(self.reports_path)
+        if 'raw_reads_r1r2' in df.columns:
+            # this is a TellSeq run: SeqCounts.csv
+            read_col = 'raw_reads_r1r2'
+            index_col = 'Sample_ID'
+        elif '# Reads' in df.columns:
+            # this is a Illumina: Demultiplex_Stats.csv
+            read_col = '# Reads'
+            index_col = 'SampleID'
+        else:
+            raise ValueError(
+                'Not sure how to check for seq counts to subsample, '
+                'please let an admin know.')
+        df = df[df[read_col] > self.MAX_READS]
+        if df.shape[0]:
+            for _, row in df.iterrows():
+                sn = row[index_col]
+                files = glob(f'{self.raw_fastq_files_path}/*/{sn}*.fastq.gz')
+                for f in files:
+                    dn = dirname(f)
+                    bn = basename(f)
+                    nbn = join(dn, bn.replace('fastq.gz', 'subsampled.gz'))
```

Comment:
Is it going to cause any problems later that the subsampled reads file doesn't end with 'fastq.gz'? I'm not familiar with the workflow but I know I've seen regexes around that expect this suffix ...